### DISCUSSION TITLE
connect fails to LAUNCHXL-­CC1352P-­2 after firmware upgrade to CC1352P2_CC2652P_launchpad_router_20210128

### DISCUSSION THREAD

**Discussion by maichai** [Koenkk/zigbee2mqtt#6634] [Devices]
The last working firmware on my LAUNCHXL-­CC1352P-­2  was CC1352P_2_20191106.hex

I updated the uniflash app on my macOS machine to 6.0.0. To be able to erase the flash the launchpad needed an automated firmware upgrade was done on the launchpad.Flash erase was successful. Writing CC1352P2_CC2652P_launchpad_router_20210128.hex was also successful according to uniflash, the verification was successful.

When connecting to the server, the launchpad used a slightly different device name, which I adapted accordingly for the docker container.

When starting the container, the connection to the launchpad fails: `Error: Failed to connect to the adapter (Error: SRSP - SYS - ping after 6000ms)`

I already tried re-attaching the launchpad to powercycle it, which did not help.

What has gone wrong, how to fix?

For now I have put CC1352P_2_20191106.hex pack on the launchpad, and the connect to this old version does work. So it is not a problem with the internal upgrade of the launchpad, it must be a problem with the CC1352P2_CC2652P_launchpad_router_20210128 firmware?

**Comment by Shred99** [comment]
> it must be a problem with the CC1352P2_CC2652P_launchpad_**router**_20210128 firmware?

Did you accidentally flash your coordinator with the router firmware?  It's easily done and will produce the symptoms that you have described.

**Comment by maichai** [comment]
yes, using the coordinator firmware it is working fine now.
